core/src/display_object.rs

```diff
@@ -1314,55 +1312,19 @@ pub trait TDisplayObject<'gc>:
         true
     }
 
-    /// Obtain the top-most non-Stage parent of the display tree hierarchy, if
-    /// a suitable object exists. If none such object exists, this function
-    /// yields an AVM1 error (which shouldn't happen in normal usage).
+    /// Obtain the top-most non-Stage parent of the display tree hierarchy.
     ///
     /// This function implements the AVM1 concept of root clips. For the AVM2
     /// version, see `avm2_root`.
-    fn avm1_root(
```

Comment:
I don't know if I'm reading this right and not sure how to observe it, but isn't this a change in behavior?
If I'm reading it right: previously, if a DO wasn't attached to a root, its `avm1_root()` would return itself, right? And after the change, it'd return the toplevel parent, whatever it is.

Reply:
Hmm, good point. But after scratching my head a bit, I think the behavior is preserved. Actually, before the change, `parent` could be `None` only in exactly 2 cases:
1. `self.lock_root()` is `true`.
2. `self.avm1_parent()` is `None`.

Once the `while let Some(p) = parent` loop is entered, `parent` was guaranteed to be `Some`:
* If `p.lock_root()`, then `parent` is `Some` as the loop condition verifies.
* Otherwise, `parent` is assigned to `grandparent` only if it's `Some`, ensuring a successive loop iteration.

After the change, these 2 cases are preserved:
1. If `self.lock_root()`, then `root` ends up being `(*self).into()`.
2. If `self.avm1_parent()` is `None`, then `root` ends up being `(*self).into()` as well.

So unless I missed something, this should be okay.